### DISCUSSION TITLE
`__typename` always returned in type even if field not present in query

### DISCUSSION THREAD

**Discussion by lmatteis** [0no-co/gql.tada#181] [Q&A] [answered]
Does anyone know why ` __typename` gets returned in my result type, even though it's not explicitly selected for in the query? I'm sure there's a reason for this but can't figure out why.

<img width="753" alt="image" src="https://github.com/0no-co/gql.tada/assets/535442/7b13f113-54e9-4845-9f1b-d727a5cf2343">

**Comment by kitten** [comment, marked as answer]
There's one where case where we now explicitly add it as an _optinal_ property like in your example. It's basically in interface/union cases, to prevent types not correctly collapsing when a `__typename` field hasn't been selected explicitly.

More info here: https://github.com/0no-co/gql.tada/pull/102

It's marked as optional though to prevent you from accidentally assuming it'll actually be present.

**Reply by lmatteis** [reply]
Thanks makes sense
